src/components/MovieList.js

```diff
@@ -2,44 +2,72 @@ import React from 'react';
 
 import MovieListItem from './MovieListItem';
 
+import {SORTING_BY_RATING, SORTING_BY_DATE} from "../constants";
+
 export default class MovieList extends React.PureComponent {
 
     constructor(props) {
         super(props);
 
         this.state = {
-            isEmpty: !(this.props.movieList && this.props.movieList.length)
+            isEmpty: !(this.props.movieList && this.props.movieList.length),
+            movieList: []
         }
     }
 
-
     static getDerivedStateFromProps(nextProps) {
```

Comment:
Note that if a parent component causes your component to re-render, this method will be called even if props have not changed. You may want to compare new and previous values if you only want to handle changes.
This article will show you best practices of using getDerivedStateFromProps hook:
https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#updating-state-based-on-props